src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -1314,8 +1337,1089 @@ public boolean mapSwitchControlValue(String value) {
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case SRT_TO_UDP_TTL:
+				createStream.setSrtToUdpTtl(Integer.parseInt(value));
+				addAdvanceControlProperties(advancedControllableProperties,
+						createNumeric(stats, streamControllingGroup + StreamControllingMetric.SRT_TO_UDP_TTL.getName(), checkForNullData(createStream.getSrtToUdpTtl().toString())));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case ENCRYPTED:
+				Boolean encrypted = mapSwitchControlValue(value);
+				removeUnusedStatsAndControlByEncrypted(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				createStream.setPassphraseSet(encrypted);
+
+				populateCreateStreamControl(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case PASSPHRASE:
+				createStream.setPassphrase(value);
+				addAdvanceControlProperties(advancedControllableProperties,
+						createText(stats, streamControllingGroup + StreamControllingMetric.PASSPHRASE.getName(), createStream.getPassphrase()));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case REJECT_UNENCRYPTED_CALLERS:
+				Boolean rejectUnencryptedCallers = mapSwitchControlValue(value);
+				createStream.setStrictMode(rejectUnencryptedCallers);
+				addAdvanceControlProperties(advancedControllableProperties, createSwitch(
+						stats, streamControllingGroup + StreamControllingMetric.REJECT_UNENCRYPTED_CALLERS.getName(), rejectUnencryptedCallers, DecoderConstant.DISABLE, DecoderConstant.ENABLE));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case CREATE:
+				performCreateStreamControl(createStream);
+				break;
+			case CANCEL:
+				createStream = defaultStream();
+				populateCreateStreamControl(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				break;
+			default:
+				throw new IllegalStateException("Unexpected value: " + controllableProperty);
+		}
+	}
+
+	/**
+	 * This method is used to perform create stream control
+	 *
+	 * @param streamInfo set of stream config info
+	 */
+	private StreamInfo performCreateStreamControl(StreamInfo streamInfo) {
+		StreamInfo streamInfoResult = new StreamInfo();
+		try {
+			if (this.authenticationCookie.getSessionID() != null) {
+				String request = streamInfo.jsonRequest();
+				if (logger.isDebugEnabled()) {
+					logger.debug(request);
+				}
+				System.out.println(buildDeviceFullPath(DecoderURL.BASE_URI + DecoderURL.STREAMS));
```

Comment:
please change to logger.debug of some other log entry if you need to track this

Reply:
I removed it.